src/packets/Serializable.ts

```diff
@@ -175,6 +166,25 @@ export abstract class Serializable {
 				);
 			}
 
+			// If the given buffer is the same as the one internally used by the view
+			// of this packet, we must assert that the new serialization would not
+			// happen in bytes where the packet currently exists.
+			// Here we must take into account the given byte offset and the byte
+			// offset of the view, the current packet length and the expected packet
+			// length after the serialization.
+			if (
+				buffer === this.view.buffer &&
+				((byteOffset >= this.view.byteOffset &&
+					byteOffset <= this.view.byteOffset + this.view.byteLength - 1) ||
+					(byteOffset + byteLength - 1 >= this.view.byteOffset &&
+						byteOffset + byteLength <=
+							this.view.byteOffset + this.view.byteLength - 1))
```

Comment:
This would better go into an util IMHO.

Reply:
This is not needed anywhere else. But if we need it somewhere else I will ofc move it into an util.